github4s/shared/src/main/scala/github4s/api/Gists.scala

```diff
@@ -56,4 +58,50 @@ class Gists[C, M[_]](
       data = NewGistRequest(description, public, files).asJson.noSpaces
     )
 
+  /**
+   * Get a single gist or a specific revision of a gist
+   *
+   * @param gistId of the gist
+   * @param sha optional sha of a revision
+   * @param headers optional user headers to include in the request
+   * @param accessToken to identify the authenticated user
+   */
+  def getGist(
+      gistId: String,
+      sha: Option[String] = None,
+      headers: Map[String, String] = Map(),
+      accessToken: Option[String] = None): M[GHResponse[Gist]] =
+    httpClient.get[Gist](
+      accessToken,
+      ("gists" :: gistId :: sha.toList).mkString("/"),
+      headers
+    )
+
+  /**
+   * Edit a gist
+   *
+   * @param gistId of the gist
+   * @param files map describing the filenames of the Gist and its contents and/or new filenames
+   * @param headers optional user headers to include in the request
+   * @param accessToken to identify the authenticated user
+   */
+  def editGist(
+      gistId: String,
+      description: String,
+      files: Map[String, Option[EditGistFile]],
+      headers: Map[String, String] = Map(),
+      accessToken: Option[String] = None): M[GHResponse[Gist]] = {
+
+    implicit val encodeEditGistFile: Encoder[EditGistFile] = {
+      deriveEncoder[EditGistFile].mapJsonObject(_.filter(e =>
+        !(e._1.equals("filename") && e._2.isNull)))
+    }
```

Comment:
You can move that to `Encoders.scala` :+1:

Reply:
I tried first, but it seems like _Encoders_ is not used here, when `toJson` is needed for implicit.

Reply:
I think you should be able to import it then, no?

Reply:
Done.